sqld/src/replication/client.rs

```diff
@@ -445,7 +446,7 @@ impl ReadReplicationHook {
                     let frame = WalFrame::decode(raw_frame.data)?;
                     debug_assert_eq!(
                         Some(frame.header.frame_id),
-                        self.wal_index_meta.map(|m| m.pre_commit_index),
+                        Some(current_offset.unwrap_or(0) + frame_count + 1),
```

Comment:
After some more thoughts I thing None should translate to -1 so it should be something like `current_offset.map(|x| x + frame_count + 1).or(Some(0)),`

Reply:
Or rather `current_offset.map(|x| x + frame_count + 1).or(Some(frame_count)),`